src/foam/nanos/auth/UserAndGroupAuthService.java

```diff
@@ -79,6 +79,32 @@ public User getCurrentUser(X x) throws AuthenticationException {
     return user;
   }
 
+  /**
+    Retrieves the agent user from the current sessions context.
+  */
+  public User getCurrentAgent(X x) throws AuthenticationException {
```

Comment:
Is this needed?

Reply:
I think it would be beneficial to provide users the ability to retrieve the current agent.

Reply:
If we know that you have to have Agent support then the AgentAuthService and AuthService could be merged. If not, then the AuthService shouldn't have this method.